vars/buildPlugin.groovy

```diff
@@ -134,12 +134,23 @@ def call(Map params = [:]) {
                         final String availableProxyProviders = configuredAvailableProxyProviders.split(',')
                         // Configure Maven settings if the requested provider is valid and available
                         if (validProxyProviders.contains(requestedProxyProvider) && availableProxyProviders.contains(requestedProxyProvider)) {
-                          echo "INFO: using artifact caching proxy from '${requestedProxyProvider}' provider."
-                          usingArtifactCachingProxy = true
-                          configFileProvider(
-                              [configFile(fileId: "artifact-caching-proxy-${requestedProxyProvider}", variable: 'MAVEN_SETTINGS')]) {
-                                infra.runMaven(mavenOptions, jdk, null, env.MAVEN_SETTINGS, addToolEnv)
-                              }
+                          boolean healthCheckOK = false
+                          final String healthCheckScript = "curl --fail https://repo.${requestedProxyProvider}.jenkins.io/healthz"
+                          if (isUnix()) {
+                            healthCheckOK = sh(script: healthCheckScript, returnStatus: true) == 0
+                          } else {
+                            healthCheckOK = bat(script: healthCheckScript, returnStatus: true) == 0
+                          }
```

Comment:
Something like that would be OK?
```suggestion
                          env.ARTIFACT_CACHING_PROXY_URL = "https://repo.${requestedProxyProvider}.jenkins.io/healthz"
                          if (isUnix()) {
                            healthCheckOK = sh(script: 'curl --fail $ARTIFACT_CACHING_PROXY_URL', returnStatus: true) == 0
                          } else {
                            healthCheckOK = bat(script: 'curl --fail $ARTIFACT_CACHING_PROXY_URL', returnStatus: true) == 0
                          }
```

Reply:
Prefer `withEnv` (scoped). `env.XXX = yyy` syntax should not normally be used.

Not a big deal, just FYI.